From fdff0e83d73991d13e54e944fa0f855c33492ae2 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:23:26 -0400
Subject: [cpufreq] intel_pstate: fix pid_reset to use fixed point values

Message-id: <1409235879-1283-113-git-send-email-prarit@redhat.com>
Patchwork-id: 89512
O-Subject: [RHEL7.1 PATCH BZ 1134639 112/185] intel_pstate: fix pid_reset to use fixed point values
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit d98d099b9fbb7bbb20a74efe5327322831c9c9a3
Author: Dirk Brandewie <dirk.j.brandewie@intel.com>
Date:   Wed Feb 12 10:01:05 2014 -0800

    intel_pstate: fix pid_reset to use fixed point values

    commit d253d2a526 (Improve accuracy by not truncating until final
    result), changed internal variables of the PID to be fixed point
    numbers. Update the pid_reset() to reflect this change.

    Signed-off-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/intel_pstate.c b/drivers/cpufreq/intel_pstate.c
index 34b8d67..7f05abe 100644
--- a/drivers/cpufreq/intel_pstate.c
+++ b/drivers/cpufreq/intel_pstate.c
@@ -153,7 +153,7 @@ static inline void pid_reset(struct _pid *pid, int setpoint, int busy,
  pid->setpoint = setpoint;
  pid->deadband  = deadband;
  pid->integral  = int_tofp(integral);
- pid->last_err  = setpoint - busy;
+ pid->last_err  = int_tofp(setpoint) - int_tofp(busy);
 }
 
 static inline void pid_p_gain_set(struct _pid *pid, int percent)
-- 
1.7.1